src/main/java/com/conveyal/gtfs/graphql/fetchers/JDBCFetcher.java

```diff
@@ -146,73 +222,94 @@ public static GraphQLFieldDefinition field (String tableName) {
-            if (parentJoinValue == null) {
-                return new ArrayList<>();
+            String inClauseString = inClauseValue == null ? null : inClauseValue.toString();
+            inClauseValues.add(inClauseString);
+            if (inClauseValue == null) {
+                return new CompletableFuture<>();
             }
         }
         Map<String, Object> arguments = environment.getArguments();
 
-        return getResults(namespace, parentJoinValues, arguments);
+        // We could select only the requested fields by examining environment.getFields(), but we just get them all.
+        // The advantage of selecting * is that we don't need to validate the field names.
+        // All the columns will be loaded into the Map<String, Object>,
+        // but only the requested fields will be fetched from that Map using a MapFetcher.
+        StringBuilder sqlBuilder = new StringBuilder();
+        sqlBuilder.append("select *");
+        return getResults(
+            getJdbcQueryDataLoaderFromContext(environment),
+            getDataSourceFromContext(environment),
+            namespace,
+            inClauseValues,
+            arguments,
+            // No additional prepared sqlStatement preparedStatementParameters, where conditions or tables are needed beyond those added by
+            // default in the next method.
+            new ArrayList<>(),
+            new ArrayList<>(),
+            new HashSet<>(),
+            sqlBuilder
+        );
     }
 
     /**
      * Handle fetching functionality for a given namespace, set of join values, and arguments. This is broken out from
-     * the standard get function so that it can be reused in other fetchers (i.e., NestedJdbcFetcher)
+     * the standard get function so that it can be reused in other fetchers (i.e., NestedJdbcFetcher).  This method
+     * ultimately returns a CompletableFuture result by calling the load method on the jdbcDataLoader argument.
+     * Therefore, this can't be directly used to get the results of a sql query.  Instead it is intended to be used only
+     * by DataFetchers that get their data by making a sql query and get their results in the format of
+     * List<Map<String, Object>>.
      */
-    List<Map<String, Object>> getResults (String namespace, List<String> parentJoinValues, Map<String, Object> arguments) {
-        // Track the parameters for setting prepared statement parameters
-        List<String> parameters = new ArrayList<>();
+    CompletableFuture<List<Map<String, Object>>> getResults (
+        DataLoader<JdbcQuery, List<Map<String, Object>>> jdbcDataLoader,
+        DataSource dataSource,
+        String namespace,
+        List<String> inClauseValues,
+        Map<String, Object> graphQLQueryArguments,
+        List<String> preparedStatementParameters,
+        List<String> whereConditions,
+        Set<String> fromTables,
+        StringBuilder sqlStatementStringBuilder
+    ) {
```

Comment:
I'm not crazy about how many arguments this function has now. I suspect @abyrd might feel the same when he reviews.

Reply:
I do agree, I generally don't like having more than 3 positional parameters. But Java doesn't have named parameters, which implies either making a new class or a new data-holder object to pass in the parameters, which are also rather ugly solutions. I know you occasionally end up with method signatures like this as part of a large refactor. It helps a lot that the Javadoc has param annotations explaining what everything does.